Defuq/Assets/Scripts/EnemyBehavior.cs

```diff
@@ -0,0 +1,67 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class EnemyBehavior : MonoBehaviour
```

Comment:
This class have too many respoinsibilities and will hard to maintian over time. I would recommend spliting the code to 3 diffrent classes: one for animation, one scriptableObject for data and one for behavoiur.

Reply:
would you recommend splitting it into 3 scripts or using different classes inside the same script?

Reply:
If you took a look now you would see I have split it into 3 different scripts